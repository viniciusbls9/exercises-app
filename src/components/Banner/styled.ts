import styled from 'styled-components';

export const BannerContainer = styled.div`
  position: relative;
  width: 100%;
  height: 0;
  padding-bottom: 56.25%; /* Aspect ratio of 16:9 */

  @media (max-width: 768px) {
    padding-bottom: 75%; /* Aspect ratio of 4:3 */
  }
`;

export const BackgroundImage = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-image: url('/young-fitness-man-studio.jpg');
  background-size: cover;
  background-position: center;
`;

export const TextOverlay = styled.div`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  text-align: center;
  color: #fff;
`;

export const Title = styled.h1`
  font-size: 32px;
  margin-bottom: 16px;

  @media (max-width: 768px) {
    font-size: 24px;
  }
`;

export const Subtitle = styled.p`
  font-size: 18px;

  @media (max-width: 768px) {
    font-size: 14px;
  }
`;