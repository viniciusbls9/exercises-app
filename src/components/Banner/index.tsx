import * as Styles from './styled'

const Banner = () => {
  return (
    <Styles.BannerContainer>
      <Styles.BackgroundImage />
      <Styles.TextOverlay>
        <Styles.Title>Find your favorite exercises</Styles.Title>
        <Styles.Subtitle>Discover amazing content and more</Styles.Subtitle>
      </Styles.TextOverlay>
    </Styles.BannerContainer>
  );
};

export default Banner;
