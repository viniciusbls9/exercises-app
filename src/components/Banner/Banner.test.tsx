import { render, screen } from '@testing-library/react';
import Banner from '.';

describe('Banner', () => {
  it('renders the title and subtitle', () => {
    render(<Banner />);
    const titleElement = screen.getByText('Find your favorite exercises');
    const subtitleElement = screen.getByText('Discover amazing content and more');
    expect(titleElement).toBeInTheDocument();
    expect(subtitleElement).toBeInTheDocument();
  });
});
