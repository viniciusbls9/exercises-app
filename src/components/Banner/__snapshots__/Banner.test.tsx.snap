// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Banner > renders the title and subtitle 1`] = `
<div>
  <div
    class="sc-aXZVg glTeJL"
  >
    <div
      class="sc-gEvEer gTBZHK"
    />
    <div
      class="sc-eqUAAy dFgdfx"
    >
      <h1
        class="sc-fqkvVR bCLAlm"
      >
        Find your favorite exercises
      </h1>
      <p
        class="sc-dcJsrY dfLJtu"
      >
        Discover amazing content and more
      </p>
    </div>
  </div>
</div>
`;
